### PR TITLE
Always use a MemberInfo for navigations if there is one matching the name

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Throw an exception when calling `SetField` would change the shadowness of a property or the name of the identifying `MemberInfo`.
Rename `GetContainingPrimaryKey` to `FindContainingPrimaryKey`
Rename `PropertyIdentity.Property` to `MemberInfo`

Part of #214

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
@ajcvickers Can you take a look as well?

File: src/EFCore.Relational/Metadata/Internal/ConstraintNamer.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ public static string GetDefaultName([NotNull] IKey key)
             var sharedTablePrincipalPrimaryKeyProperty = key.Properties[0].FindSharedTableRootPrimaryKeyProperty();
             if (sharedTablePrincipalPrimaryKeyProperty != null)
             {
-                return sharedTablePrincipalPrimaryKeyProperty.GetContainingPrimaryKey().Relational().Name;
```
Comment (ajcvickers) [inline src/EFCore.Relational/Metadata/Internal/ConstraintNamer.cs]: Is this name change for an shipped public method? If so, is it worth breaking for to change Get to Find?

File: src/EFCore/Metadata/Builders/CollectionNavigationBuilder.cs [hunk 2]
```diff
@@ -156,12 +156,12 @@ private InternalRelationshipBuilder WithOneBuilder(PropertyIdentity reference)
 
             return referenceName != null
                    && RelatedEntityType != foreignKey.DeclaringEntityType
-                ? reference.Property == null && CollectionProperty == null
```
Comment (ajcvickers) [inline src/EFCore/Metadata/Builders/CollectionNavigationBuilder.cs]: Likewise on this change?

Comment (AndriySvyryd) [reply src/EFCore/Metadata/Builders/CollectionNavigationBuilder.cs]: `PropertyIdentity` is `Internal`

File: src/EFCore.Relational/Metadata/Internal/ConstraintNamer.cs (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore.Relational/Metadata/Internal/ConstraintNamer.cs]: I'll obsolete the old one.

**Pull request merged**